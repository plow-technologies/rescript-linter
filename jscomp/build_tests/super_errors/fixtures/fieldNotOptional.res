type r = {nonopt: int, opt?: string}

let v = {nonopt: ?3, opt: ?None}

let f = r =>
  switch r {
  | {nonopt: ?_, opt: ?_} => true
  }

type inline = A({nonopt: int, opt?: string})

let vi = A({nonopt: ?3, opt: ?None})

let fi = a =>
  switch a {
  | A ({nonopt: ?_, opt: ?_}) => true
  }