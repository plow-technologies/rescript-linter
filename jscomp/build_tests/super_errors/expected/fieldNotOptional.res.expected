
  [1;31mWe've found a bug for you![0m
  [36m/.../fixtures/fieldNotOptional.res[0m:[2m3:19[0m

  1 [2mâ”‚[0m type r = {nonopt: int, opt?: string}
  2 [2mâ”‚[0m 
  [1;31m3[0m [2mâ”‚[0m let v = {nonopt: ?[1;31m3[0m, opt: ?None}
  4 [2mâ”‚[0m 
  5 [2mâ”‚[0m let f = r =>

  Field [1;33mnonopt[0m is not optional in type r. Use without ?